compiler/ast/src/passes/visitor.rs

```diff
@@ -114,7 +114,15 @@ pub trait ExpressionVisitor<'a> {
         Default::default()
     }
 
-    fn visit_struct_init(&mut self, _input: &'a StructExpression, _additional: &Self::AdditionalInput) -> Self::Output {
+    fn visit_struct_init(&mut self, input: &'a StructExpression, additional: &Self::AdditionalInput) -> Self::Output {
+        let StructExpression { name, members, .. } = input;
+        self.visit_identifier(name, additional);
```

Comment:
Should we be visiting the identifier? If yes, then should we visit function names and other similar constructs as well?
The visitor doesn't have a canonical definition but it could be worth setting one.